### ISSUE TITLE
Too many tokens in summaries

### ISSUE THREAD

**Issue by Klaudioz** [fjrdomingues/autopilot#61]
I just got this error:

`Aborting. Too many tokens in summaries. 4593 Max allowed: 3000
/Users/canalescl/personal/autopilot/ui.js:17
    throw new Error(message)`

is anything I can do?. I have GPT-4 access.

**Comment by fjrdomingues** [comment]
We have a plan to solve that limitation #31 
Since you have access to gpt4 you can be unblocked for now by changing the hardcoded value. Look for `maxSummaryTokenCount` and increase the value

**Comment by Klaudioz** [comment]
Thank you. I updated it to 8192, but I got the following error now:

```
Reply: undefined
failed to parse JSON SyntaxError: "undefined" is not valid JSON
    at JSON.parse (<anonymous>)
    at jsonParseWithValidate (/Users/canalescl/personal/autopilot/modules/gpt.js:87:17)
    at getRelevantFiles (/Users/canalescl/personal/autopilot/agents/getFiles.js:36:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runAgent (/Users/canalescl/personal/autopilot/ui.js:40:10)
    at async main (/Users/canalescl/personal/autopilot/ui.js:75:19) undefined
/Users/canalescl/personal/autopilot/modules/gpt.js:90
    throw new Error('Invalid JSON');
          ^

Error: Invalid JSON
    at jsonParseWithValidate (/Users/canalescl/personal/autopilot/modules/gpt.js:90:11)
    at getRelevantFiles (/Users/canalescl/personal/autopilot/agents/getFiles.js:36:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runAgent (/Users/canalescl/personal/autopilot/ui.js:40:10)
    at async main (/Users/canalescl/personal/autopilot/ui.js:75:19)
```

**Comment by hatkyinc2** [comment]
These Errors mean that the model returned something unexpected (or it was a bad prompt at the time)
You can also try interactive mode `node ui -i` to see the process of what we receive and retry...
We are also hacking on this intensively multiple times a day and it's < weeks old, so please do try again when you got some extra time @Klaudioz

**Comment by Klaudioz** [comment]
I just updated to the last version and I got the same.

Also to change the number of tokens, I can see a lot of hardcoded references to `3000`

```sh
~/p/autopilot (main|✔) $ find . -type f | xargs grep -Hnw --color "3000"
./createSummaryOfFiles.js:44:      if (countTokens(file) > 3000) {
./node_modules/langchain/dist/memory/motorhead_memory.js:19:            value: 3000
./node_modules/langchain/dist/memory/motorhead_memory.cjs:22:            value: 3000
./node_modules/langchain/dist/chains/combine_docs_chain.cjs:117:            value: 3000
./node_modules/langchain/dist/chains/combine_docs_chain.js:113:            value: 3000
./benchmarks/files/ui.js:8:const maxSummaryTokenCount = 3000;
./modules/summaries.js:9:const maxSummaryTokenCount = 3000;
```

**Comment by ofir-petrushka-lat** [comment]
Ignore the node_modules directly it's external code

**Comment by hatkyinc2** [comment]
@Klaudioz If you want to try the `edge` branch, it should fix the summary limit
https://github.com/fjrdomingues/autopilot/tree/edge

**Comment by hatkyinc2** [comment]
We are now chunking summaries, this limit no longer exists.
@Klaudioz If you get another issue running autopilot, do open another issue.
